### PR TITLE
[MPIR-429] disable plugin processing when building projects

### PR DESCRIPTION
**Pull request by hboutemy**
https://issues.apache.org/jira/browse/MPIR-429

### CODE REVIEW

**Comment by michael-o** [comment]
I still don't understand why this does fix the issue?

**Comment by hboutemy** [comment]
analysis done https://issues.apache.org/jira/browse/MDEP-630 : issue happens in Maven's lifecycle bindings injector = something that is optional, used when the effective model is expected to be used for real building a Maven project, with its plugins

but in our reporting case, dependencies are just dependencies consumed, we don't expect them to be built: then anything related to plugins may not even be right in the pom (when it is generated by Gradle, sbt or Ant, for example)

consumer pom.xml should be read with lax schema checks and disabling plugins injection to be the most tolerant

**Comment by michael-o** [comment]
Nice one. Does it make sense to revert MPIR-374?

**Comment by hboutemy** [comment]
I think that MPIR-374 still makes sense: no need to add eventual noise

**Review by michael-o — APPROVED** [review]

File: src/main/java/org/apache/maven/report/projectinfo/PluginManagementReport.java [hunk 1]
```diff
@@ -206,6 +206,7 @@ private void renderSectionPluginManagement()
 
             ProjectBuildingRequest buildRequest = new DefaultProjectBuildingRequest( buildingRequest );
             buildRequest.setRemoteRepositories( project.getPluginArtifactRepositories() );
+            buildingRequest.setProcessPlugins( false );
```
Comment (michael-o) [inline src/main/java/org/apache/maven/report/projectinfo/PluginManagementReport.java:209]: buildRequest or buildingRequest?

File: src/main/java/org/apache/maven/report/projectinfo/PluginsReport.java [hunk 2]
```diff
@@ -192,6 +192,7 @@ private void renderSectionPlugins( boolean isPlugins )
 
             ProjectBuildingRequest buildRequest = new DefaultProjectBuildingRequest( buildingRequest );
             buildRequest.setRemoteRepositories( project.getPluginArtifactRepositories() );
+            buildingRequest.setProcessPlugins( false );
```
Comment (michael-o) [inline src/main/java/org/apache/maven/report/projectinfo/PluginsReport.java:195]: buildRequest or buildingRequest?

Comment (hboutemy) [reply src/main/java/org/apache/maven/report/projectinfo/PluginsReport.java]: thanks for the review...

**Pull request merged**